libAfterImage: unvendor dependencies

veprbl opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Fixes: #123408

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

veprbl committed:
libAfterImage: unvendor dependencies

veprbl committed:
libAfterImage: don't use weird mmx flags

veprbl committed:
libAfterImage: build shared libs by default

r-rmcgibbo commented:
Result of `nixpkgs-review pr 123638` at aec53db2 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages skipped due to time constraints:</summary>
  <ul>
    <li>hepmc3</li>
    <li>python38Packages.hepmc3</li>
    <li>python39Packages.hepmc3</li>
    <li>root</li>
    <li>yoda-with-root</li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>libAfterImage</li>
    <li>tests.trivial</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl21` is a deprecated license, please check if project uses `lgpl21Plus` or `lgpl21Only` and change `meta.license` accordingly.

Near pkgs/development/libraries/libAfterImage/default.nix:73:5:
```
   |
73 |     license = licenses.lgpl21;
   |     ^
```
</li>
  </ul>
</details>

veprbl committed:
libAfterImage: compile against libX11

This is needed to not have X_DISPLAY_MISSING for asimage2pixmap function
to be implemented:
https://root-forum.cern.ch/t/root-xpm-error-on-ubuntu-revisited/4552/10

veprbl commented:
Result of `nixpkgs-review pr 123638` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>yoda-with-root</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>libAfterImage</li>
    <li>python38Packages.hepmc3</li>
    <li>python39Packages.hepmc3</li>
    <li>root</li>
  </ul>
</details>

veprbl commented:
Result of `nixpkgs-review pr 123638` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>tests.trivial</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>libAfterImage</li>
    <li>python38Packages.hepmc3</li>
    <li>python39Packages.hepmc3</li>
    <li>root</li>
    <li>yoda-with-root</li>
  </ul>
</details>
